test/typed_struct_test.exs

```diff
@@ -20,15 +20,49 @@ defmodule TypedStructTest do
       def enforce_keys, do: @enforce_keys
     end
 
-  {:module, _name, bytecode_opaque, _exports} =
-    defmodule OpaqueTestStruct do
+  {:module, _name, bytecodepublic2, _exports} =
+    defmodule TestStructPublic2 do
+      use TypedStruct
+
+      typedstruct visibility: :public do
+        field :int, integer()
+        field :string, String.t()
+        field :string_with_default, String.t(), default: "default"
+        field :mandatory_int, integer(), enforce: true
+      end
+
+      def enforce_keys, do: @enforce_keys
+    end
+
+  {:module, _name, bytecode_opaque1, _exports} =
+    defmodule OpaqueTestStruct1 do
       use TypedStruct
 
       typedstruct opaque: true do
         field :int, integer()
       end
     end
 
+  {:module, _name, bytecode_opaque2, _exports} =
+    defmodule OpaqueTestStruct2 do
+      use TypedStruct
+
+      typedstruct visibility: :opaque do
+        field :int, integer()
+      end
+    end
+
+  {:module, _name, bytecode_private, _exports} =
+    defmodule PrivateTestStruct do
+      use TypedStruct
+
+      typedstruct visibility: :private do
+        field :int, integer()
+      end
+
+      @opaque tt :: t  # Needed so that the compiler doesn't remove unused private type t()
```

Comment:
This should be `@opaque tt() :: t()` to be consistent with the style used in the library.